packages/gatsby/src/schema/schema.js

```diff
@@ -1046,9 +1055,15 @@ const addImplicitConvenienceChildrenFields = ({
         )
       }
     }
-
-    typeComposer.addFields(createChildrenField(typeName))
-    typeComposer.addFields(createChildField(typeName))
+    // Set `@childOf` extension automatically
+    // This will cause convenience children fields like `childImageSharp`
+    // to be added in `addConvenienceChildrenFields` method.
+    // Also required for proper printing of the `@childOf` directive in the snapshot plugin
+    if (!childOfExtension?.types) {
+      childOfExtension = { types: [] }
+    }
```

Comment:
Question - `@childOf` can have either `types` or `mimeTypes` set - https://github.com/gatsbyjs/gatsby/blob/8ff6245f44882d891bc6f291363019047cbdb55f/packages/gatsby/src/schema/extensions/index.js#L58-L85

It has defaultValue set to empty array - but what happens when extension is set programmatically with `schema.buildObjectType` - does this also use that `defaultValue`?

I just wonder if this check might ovverride some extension that omit to define `types`, but do use `mimeTypes`

Reply:
Alternatively - should newly `childExtension` created here set `mimeTypes` to empty array? too :)

Reply:
Good catch 👍 Added `mimeTypes` there.

Reply:
Actually, this causes printer snapshot failures because now `mimeTypes: []` is printed anytime `childOf` is used. I've updated the code. The new version won't add `mimeTypes` by default but will preserve it if it was set for this extension somewhere else.